### PR TITLE
Bugfix #1124 When CSDL filter and http resource supplied for HIDI input, it fails because ApplyFilter expects a local file but not a URL address

### PR DESCRIPTION
**Pull request by MazZzDaI**
Fix of the #1124 issue.

Overload added for _Microsoft.OpenApi.Hidi/OpenApiService.cs/ApplyFilter_ to accept both string and stream variables.

### CODE REVIEW

**Comment by MazZzDaI** [comment]
@microsoft-github-policy-service agree [company="Olk inc."]

File: src/Microsoft.OpenApi.Hidi/OpenApiService.cs [hunk 1]
```diff
@@ -98,7 +98,7 @@ CancellationToken cancellationToken
                         if (!string.IsNullOrEmpty(csdlFilter))
                         {
                             XslCompiledTransform transform = GetFilterTransform();
-                            stream = ApplyFilter(csdl, csdlFilter, transform);
+                            stream = csdl.StartsWith("http") ? ApplyFilter(stream, csdlFilter, transform) : ApplyFilter(csdl, csdlFilter, transform);
```
Comment (baywet) [inline src/Microsoft.OpenApi.Hidi/OpenApiService.cs:101]:
```suggestion
                            stream = csdl.StartsWith("http", StringComparison.OrdinalIgnoreCase) ? ApplyFilter(new StreamReader(stream), csdlFilter, transform) : ApplyFilter(new StreamReader(csdl), csdlFilter, transform);
```

File: src/Microsoft.OpenApi.Hidi/OpenApiService.cs [hunk 2]
```diff
@@ -235,20 +235,32 @@ private static XslCompiledTransform GetFilterTransform()
...
+            StreamReader inputReader = new(csdl);
+            return ApplyFilterForStreamReader(inputReader, entitySetOrSingleton, transform);
+        }
+
+        private static Stream ApplyFilter(Stream csdlStream, string entitySetOrSingleton, XslCompiledTransform transform)
+        {
+            StreamReader inputReader = new(csdlStream);
+            return ApplyFilterForStreamReader(inputReader, entitySetOrSingleton, transform);
+        }
+
```
Comment (baywet) [inline src/Microsoft.OpenApi.Hidi/OpenApiService.cs:263]:
```suggestion
```

File: src/Microsoft.OpenApi.Hidi/OpenApiService.cs [hunk 3]
```diff
@@ -235,20 +235,32 @@ private static XslCompiledTransform GetFilterTransform()
             return transform;
         }
 
-        private static Stream ApplyFilter(string csdl, string entitySetOrSingleton, XslCompiledTransform transform)
+        private static Stream ApplyFilterForStreamReader(StreamReader inputReader, string entitySetOrSingleton, XslCompiledTransform transform)
```
Comment (baywet) [inline src/Microsoft.OpenApi.Hidi/OpenApiService.cs:238]:
```suggestion
        private static Stream ApplyFilter(StreamReader inputReader, string entitySetOrSingleton, XslCompiledTransform transform)
```

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution, here are a couple of comments to help improve it

File: src/Microsoft.OpenApi.Hidi/OpenApiService.cs (hunk 1 above)
Comment (darrelmiller) [reply src/Microsoft.OpenApi.Hidi/OpenApiService.cs:101]: The original code here seems wrong.  GetStream does all the logic to determine if stream  is coming from a file or via HTTP.  Currently the code is getting the stream and then ignoring it if there is a filter.  The solution should be to change ApplyFilter to just accept the stream that comes from GetStream.

**Comment by darrelmiller** [comment]
I was able to resolve this issue in PR #1113 in this commit https://github.com/microsoft/OpenAPI.NET/pull/1113/commits/7aac03ffd6f5e1fbe6d6c8cb07a280c1bacc47c4  
I also added support for using CSDL with the show command.

**Comment by darrelmiller** [comment]
Thank you @MazZzDaI for identifying this issue.  PR #1113 has the fixes you are looking for and other related improvements.

**Pull request closed without merging**